sources/packages/backend/apps/api/src/route-controllers/application/application.controller.service.ts

```diff
@@ -920,6 +923,9 @@ export class ApplicationControllerService {
       payload,
       application.offeringIntensity,
     );
+    // Inject the indicator for change request submission.
+    payload.data.isChangeRequestApplication =
+      options?.isChangeRequestSubmission ?? false;
```

Comment:
Is this association not a responsibility of the `validateSubmitApplicationData`.
My understanding from the `validateSubmitApplicationData` is that it must resolve any payload-specific data-related task.

Reply:
I too started in the same way, but did the injection here trying to avoid cascading the optional flag to `validateSubmitApplicationData`. I can change.

Reply:
Changed.